crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -2187,17 +2207,40 @@ pub trait CheckpointServiceNotify {
     fn notify_checkpoint(&self) -> SuiResult;
 }
 
-/// This is a service used to communicate with other pieces of sui(for ex. authority)
+enum CheckpointServiceState {
+    Unstarted((CheckpointBuilder, CheckpointAggregator)),
+    Started,
+}
+
+impl CheckpointServiceState {
+    fn take(&mut self) -> (CheckpointBuilder, CheckpointAggregator) {
```

Comment:
could we have a more desciptive name for this than `take`? I think the behavior (about marking it started) is surprising given the fn name

Reply:
fair enough - i named it take because I was just making a more readable version of an Option. renamed to `take_unstarted`